### ISSUE TITLE
NFR: Change Truck Configurator Integration to Enhanced Embed Solution

### ISSUE THREAD

**Issue by ZoeBeykirch** [hlxsites/vg-volvotrucks-us#704]
### Context/Description

As a Volvo Trucks website owner, I want to replace the current iFrame integration for the truck configurator with the improved embed solution, so that we can enhance loading performance and have more analytics capabilities, replicating the successful embed implementation used for the Mack Trucks launch.

### Acceptance Criteria

**User AC:**
- [ ] Ensure that all configurator functionalities and user settings are preserved and fully operational.

**Non-functional AC:**
- [ ] Replace the existing iFrame integration with the new embed solution for USA and Canadian sites:
   -  https://www.volvotrucks.us/truck-builder
   - https://www.volvotrucks.ca/en-ca/truck-builder
   - https://www.volvotrucks.ca/fr-ca/truck-builder
- [ ] Performance loading times benchmarks should meet or exceed those achieved during the Mack Trucks launch.
- [ ] GA4 analytics tracking should work with the new embed solution to monitor user interactions.

### Note

The embed solution should incorporate the same improvements and features successfully implemented during the Mack Trucks launch:
https://github.com/hlxsites/vg-macktrucks-com/issues/579
https://github.com/hlxsites/vg-macktrucks-com/issues/591

**Comment by cincplug** [comment]
We will need to have CORS issue resolved on RapidImages side in order to have blocks rendered on our side. Kalle is informed about the issue and he'll contact their backend

**Comment by cincplug** [comment]
Configurator functionality is now embedded in our page instead of an iframe, using the same procedure like previously on macktrucks.com. It can be tested on this draft page: 
https://develop--vg-volvotrucks-us--hlxsites.hlx.page/drafts/luka/truck-builder, 
compared to: https://develop--vg-volvotrucks-us--hlxsites.hlx.page/drafts/luka/truck-builder,
on which the current iframe solution is still in use.

Text colors are a bit different in nuances between these two implementations, because color definitions in our stylesheets now override some of those defined by Rapid Images. I recommend keeping our definitions for better consistency with the rest of volvotrucks site.

Anchor part of the URL in the address bar now changes to different anchors, which wasn't visible before because the URL was within the iframe

Here is the modified [sharepoint file for the metadata](https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7BCA406094-B598-4EBD-B6CB-EC90230CD854%7D&file=truck-builder.docx&action=default&mobileredirect=true), I'll do these modifications in original file after testing

**Comment by girishdigrajkar** [comment]
Validated and shared observations with Dev team for fixes; reassigning it to 'In Development' status.

https://ncchat.slack.com/archives/C073XMHHRNK/p1726831419233669

**Comment by verareljic** [comment]
2 SPs delivered

**Comment by mollyaljenkins** [comment]
Update Sept 23: Moved AC to out of scope to simplify this issue. Adjusted to 2 SP for the remaining effort

**Comment by cincplug** [comment]
Some issues found by @girishdigrajkar were fixed, I commented it in the thread started by Girish, but I'm still working on one before putting it back into testing

**Comment by BeckyMedlin** [comment]
@mollyaljenkins How will the team address the out of scope items? 
- Performance loading times benchmarks should meet or exceed those achieved during the Mack Trucks launch.
- GA4 analytics tracking should work with the new embed solution to monitor user interactions.

**Comment by mollyaljenkins** [comment]
@BeckyMedlin  These will be future enhancements if they are seen as still needed as discussed with @cincplug @ZoeBeykirch

**Comment by BeckyMedlin** [comment]
@williebender This ticket is blocked as we are awaiting some fixes from Rapid Images and they need budget approval from Volvo.

**Comment by cincplug** [comment]
We are dependent on RapidImages to adjust the "Copy link" button and "Start a new configuration" / "Back to configurator" so that they would lead to correct location now that they are out of iframe context, since these buttons are within their bundle and we have no control over them

**Comment by BeckyMedlin** [comment]
Moved out of current sprint due to blocker w/ Rapid Images.

**Comment by BeckyMedlin** [comment]
Per @williebender PO for Rapid Image work has been issued. Waiting on Rapid to complete their efforts.
